api/environments/identities/traits/models.py

```diff
@@ -3,14 +3,31 @@
 from django.core.exceptions import ObjectDoesNotExist
 from django.db import models
 from django.utils.encoding import python_2_unicode_compatible
-from simple_history.models import HistoricalRecords
+from django_lifecycle import AFTER_DELETE, AFTER_SAVE, LifecycleModel, hook
 
 from environments.identities.traits.exceptions import TraitPersistenceError
 from environments.models import BOOLEAN, FLOAT, INTEGER, STRING
 
 
+class TriatQuerySet(models.QuerySet):
+    def delete(self, *args, **kwargs):
+        # collect all the identities that we need to update
+        identitie_to_delete = [trait.identity for trait in self]
+        super().delete(*args, **kwargs)
+        # update all the identities
+        if identitie_to_delete:
+            identitie_to_delete[0].bulk_send_to_dynamodb(identitie_to_delete)
+
+    def bulk_create(self, *args, **kwargs):
+        super().bulk_create(*args, **kwargs)
+        # Since we only bulk create traits for the same identity
+        # It is safe to pick any element from the qs
+        if self:
```

Comment:
Does this work? My understanding is that an empty queryset still evaluates to True - don't we want to check `.count() >= 1` maybe?

Reply:
Yep. it does work just like an empty string.
here is the implementation: 
```

    def __bool__(self):
        self._fetch_all()
        return bool(self._result_cache)


```